static/js/login_or_signup.js

```diff
@@ -51,103 +50,87 @@ $(function() {
+                $field.keypress((event) => {
+                    var fieldHasValue = $.trim($field.val()).length > 0;
+
+                    if (checkKeypressIsSpace(event) && !fieldHasValue) {
+                        return false; // cancels keypress event
+                    }
+                });
+
+                // remove trailing spaces on blur
+                $field.blur(() => {
+                    var value = $field.val();
+                    var trimmedValue = $.trim(value);
+
+                    $field.val(trimmedValue);
+                });
+            }
+        });
     };
 
     installFieldValidators(signupFormMetadata);
     installFieldValidators(loginFormMetadata);
 
     var installSubmitClickHandler = function(formMetadata, postFunction) {
-        formMetadata.$form.find('button').click(
-            () => {
-                if (checkFormValidity(formMetadata.$form, formMetadata.fields)) {
-                    var formData = getFormData(formMetadata.$form, formMetadata.fields);
-                    var formDataAsJsonString = JSON.stringify(formData);
+        formMetadata.$form.find('button').click(() => {
+            if (checkFormValidity(formMetadata.$form, formMetadata.fields)) {
+                var formData = getFormData(
+                    formMetadata.$form,
+                    formMetadata.fields);
 
-                    postFunction(formDataAsJsonString)
+                var formDataAsJsonString = JSON.stringify(formData);
 
-                } else if (!formMetadata.submitHasBeenClicked) {
-                    formMetadata.submitHasBeenClicked = true;
+                postFunction(formDataAsJsonString)
 
-                    populateValidationMessages(formMetadata.$form, formMetadata.fields);
-                    touchAllFields(formMetadata.$form, formMetadata.fields);
-                }
-            });
-    }
+            } else if (!formMetadata.submitHasBeenClicked) {
+                formMetadata.submitHasBeenClicked = true;
 
-    installSubmitClickHandler(
-        signupFormMetadata, 
-        (formDataAsJsonString) => {
-            $.post(
-                '/user',
-                formDataAsJsonString,
-                (userId) => {
-                    alert('Created User with id: ' + userId.value);
-                },
-                'json');
-        });
+                populateValidationMessages(
+                    formMetadata.$form,
+                    formMetadata.fields);
 
-    installSubmitClickHandler(
-        loginFormMetadata, 
-        (formDataAsJsonString) => {
-            $.post(
-                '/session', 
-                formDataAsJsonString, 
-                (response) => {
-                    alert(response);
-                }, 
-                'text');
+                touchAllFields(formMetadata.$form, formMetadata.fields);
+            }
         });
+    }
+
+    installSubmitClickHandler(signupFormMetadata, (formDataAsJsonString) => {
```

Comment:
Do you want to rename these as attach.... Or do you want me to do that in a different pull request?

Reply:
I'll make that change